app/auth/signup/email/action.ts

```diff
@@ -0,0 +1,22 @@
+'use server'
+import { cookies } from 'next/headers'
+import { redirect } from 'next/navigation'
+import { createClient } from '../../supabase'
+import type { SignUpSchema } from './page'
+
+export const signUp = async ({
```

Comment:
Once the sign-up is completed, we should ensure that the user's record is saved in the Supabase Database (public schema users table). :eyes:

For example, Supabase provides a user trigger.
https://supabase.com/docs/guides/auth/managing-user-data#using-triggers